### PR TITLE
Add Thermostats and Climate Settings

### PR DESCRIPTION
**Pull request by abimaelmartell**
Add the following resources

- Thermostats
- Climate Setting Schedules

### CODE REVIEW

**Review by codetheweb — APPROVED** [review]

**Review by dawnho — APPROVED** [review]

File: seamapi/climate_setting_schedules.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+          Intial seam class
+        """
+
+        self.seam = seam
+
+    @report_error
+    def list(
+        self,
+        device: Union[DeviceId, Device],
+    ) -> List[Device]:
```
Comment (andrii-balitskyi) [inline seamapi/climate_setting_schedules.py]:
```suggestion
    ) -> List[ClimateSettingSchedule]:
```

File: seamapi/climate_setting_schedules.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+        )
+        climate_setting_schedules = res["climate_setting_schedules"]
+
+        return [ClimateSettingSchedule.from_dict(d) for d in climate_setting_schedules]
+
+    @report_error
+    def get(
+        self,
+        climate_setting_schedule: Union[ClimateSettingScheduleId, ClimateSettingSchedule]
+    ) -> Device:
```
Comment (andrii-balitskyi) [inline seamapi/climate_setting_schedules.py]:
```suggestion
    ) -> ClimateSettingSchedule:
```

File: seamapi/climate_setting_schedules.py [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+        automatic_heating_enabled: Optional[bool] = None,
+        automatic_cooling_enabled: Optional[bool] = None,
+        hvac_mode_setting: Optional[str] = None,
+        cooling_set_point_celsius: Optional[float] = None,
+        heating_set_point_celsius: Optional[float] = None,
+        cooling_set_point_fahrenheit: Optional[float] = None,
+        heating_set_point_fahrenheit: Optional[float] = None,
+        manual_override_allowed: Optional[bool] = None,
+        schedule_type: Optional[str] = None,
+    ) -> bool:
```
Comment (andrii-balitskyi) [inline seamapi/climate_setting_schedules.py]:
```suggestion
    ) -> ClimateSettingSchedule:
```

File: seamapi/climate_setting_schedules.py [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+            json=params,
+        )
+
+        return True
+
+    @report_error
+    def delete(
+        self,
+        climate_setting_schedule: Optional[Union[str, ClimateSettingSchedule]]
+    ) -> bool:
```
Comment (andrii-balitskyi) [inline seamapi/climate_setting_schedules.py]:
```suggestion
    ) -> None:
```

File: seamapi/routes.py [hunk 5]
```diff
@@ -20,6 +22,8 @@ def __init__(self):
       self.access_codes = AccessCodes(seam=self)
       self.action_attempts = ActionAttempts(seam=self)
       self.noise_sensors = NoiseSensors(seam=self)
+      self.thermostats = Thermostats(seam=self)
+      self.climate_setting_schedules = ClimateSettingSchedules(seam=self)
```
Comment (andrii-balitskyi) [inline seamapi/routes.py]: I think this shouldn't be available here. `climate_setting_schedules` should be accessible under thermostats eg. `seam.thermostats.climate_setting_schedules.get`.

File: seamapi/thermostats.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        self.seam.make_request(
+            "POST",
+            "/thremostats/update",
+            json=update_payload,
+        )
+
+        return True
+
+    @report_error
+    def delete(self, device: Union[DeviceId, Device]) -> bool:
```
Comment (andrii-balitskyi) [inline seamapi/thermostats.py:186]:
```suggestion
    def delete(self, device: Union[DeviceId, Device]) -> None:
```

File: seamapi/types.py [hunk 7]
```diff
@@ -193,6 +194,38 @@ def from_dict(nt: Dict[str, Any]):
...
+@dataclass
+class ClimateSetting:
+    automatic_heating_enabled: Optional[bool]
+    automatic_cooling_enabled: Optional[bool]
+    hvac_mode_setting: Optional[str]
+    cooling_set_point_celsius: Optional[float]
+    heating_set_point_celsius: Optional[float]
+    cooling_set_point_fahrenheit: Optional[float]
+    heating_set_point_fahrenheit: Optional[float]
+    manual_override_allowed: Optional[bool]
```
Comment (andrii-balitskyi) [inline seamapi/types.py:208]: there's also `is_set_on_device` property if I'm not mistaken

File: seamapi/thermostats.py [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+        res = self.seam.make_request("GET", "/thermostats/get", params=params)
+        json_thermostat = res["thermostat"]
+        return Device.from_dict(json_thermostat)
+
+    @report_error
+    def update(
+        self,
+        device: Union[DeviceId, Device],
+        name: Optional[str] = None,
```
Comment (andrii-balitskyi) [inline seamapi/thermostats.py]: I don't see a `name` param on seam-connect [update endpoint](https://github.com/seamapi/seam-connect/blob/018614788c73e7fb34f3df0bb3c9f222d4823ed6/pages/api/public/thermostats/update.ts#L8-L9). I guess it should've been added

File: tests/thermostats/test_thermostats.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+from seamapi import Seam
+from tests.fixtures.run_nest_factory import run_nest_factory
+
+def test_thermostats(seam: Seam):
+    run_nest_factory(seam)
+
+    # Test List
+    thermostats = seam.thermostats.list()
+
+    print(thermostats)
```
Comment (andrii-balitskyi) [inline tests/thermostats/test_thermostats.py]:
```suggestion
```

File: tests/thermostats/test_climate_setting_schedules.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    climate_setting_schedule = seam.thermostats.climate_setting_schedules.get(
+        climate_setting_schedule=climate_setting_schedule,
+    )
+
+    assert climate_setting_schedule.name == "Vacation Setting 2"
+
+    # Test Delete
+    result = seam.thermostats.climate_setting_schedules.delete(
+        climate_setting_schedule=climate_setting_schedule,
+    )
```
Comment (andrii-balitskyi) [inline tests/thermostats/test_climate_setting_schedules.py:55]: We can also try to get that `climate_setting_schedule` to see if it was actually deleted

File: tests/thermostats/test_thermostats.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    assert thermostat.device_type == "nest_thermostat"
+
+    # Test Update
+    result = seam.thermostats.update(
+        device=thermostat,
+        default_climate_setting={
+            "hvac_mode_setting": "cool",
+            "cooling_set_point_celsius": 20,
+        }
+    )
```
Comment (andrii-balitskyi) [inline tests/thermostats/test_thermostats.py:27]: Here we can also get the `thermostat` to see if it was actually updated

**Review by andrii-balitskyi — APPROVED** [review]
minor notes, but generally looks good!

File: seamapi/climate_setting_schedules.py [hunk 12]
```diff
@@ -0,0 +1,344 @@
...
+        )
+        climate_setting_schedules = res["climate_setting_schedules"]
+
+        return [ClimateSettingSchedule.from_dict(d) for d in climate_setting_schedules]
+
+    @report_error
+    def get(
+        self,
+        climate_setting_schedule: Union[ClimateSettingScheduleId, ClimateSettingSchedule]
+    ) -> ClimateSettingSchedule:
```
Comment (andrii-balitskyi) [inline seamapi/climate_setting_schedules.py:115]: /get endpoint also accepts `device_id`

Comment (abimaelmartell) [reply seamapi/climate_setting_schedules.py:115]: we are actually removing this param from that endpoint

File: tests/thermostats/test_thermostats.py (hunk 11 above)
Comment (abimaelmartell) [reply tests/thermostats/test_thermostats.py:27]: the thing is that this happens async, so we would need to wait for the event to come in, this tests would only make sure that the endpoint is taking the data, and returning success

**Pull request merged**